dpctl/tests/test_usm_ndarray_ctor.py

```diff
@@ -1282,3 +1284,24 @@ def test_common_arg_validation():
         dpt.ones_like(X)
     with pytest.raises(TypeError):
         dpt.full_like(X, 1)
+    with pytest.raises(TypeError):
+        dpt.eye(4, k=1.2)
+
+
+@pytest.mark.parametrize("shapes", [(0,), (1,), (7,), (6, 1), (3, 9), (10, 5)])
+@pytest.mark.parametrize("k", np.arange(-4, 5, 1))
+@pytest.mark.parametrize("orders", ["C", "F"])
+def test_eye(shapes, k, orders):
```

Comment:
Test should be exercising all supported array data types. Remember that some HW, like Iris Xe, does not support 64-bit floating point type, and hence does not support `complex128` by extension.